### PR TITLE
Added a server info endpoint providing version number details

### PR DESCRIPTION
**Pull request by Vektrat**
I think this will help with reporting errors and the number of confusions that have sprouted lately with the increased dev activity

@dgarcia360 probably of interest for you for documentation, @evias also a "tool" you might find useful

### CODE REVIEW

**Comment by dgarcia360** [comment]
Developers could find confusing to have both ``node/info`` and ``diagnostic/serverInfo`` endpoints.

Consider one of the following improvements:

1. Move the new endpoint to ``/node/version``. To have a better experience, we could access the versions using an identifier instead of looking an ordered array.

    {
            "node": [nodeVersion](https://github.com/nemtech/catapult-server/blob/master/resources/config-node.properties#L51),
            "rest": restVersion,
            "sdk":sdkVersion
    }

2. Reuse``node/info`` endpoint. This already has a version field, which refers to the version defined in the node configuration parameters. The downside is that this change would break compatibility.

File: catapult-sdk/src/model/ModelSchemaBuilder.js [hunk 1]
```diff
@@ -153,6 +153,10 @@ class ModelSchemaBuilder {
 				communicationTimestamps: { type: ModelType.object, schemaName: 'communicationTimestamps' }
 			},
 
+			serverInfo: {
+
```
Comment (Jaguar0625) [inline catapult-sdk/src/model/ModelSchemaBuilder.js]: match following `storageInfo ` format and don't have blank line

File: rest/src/routes/diagnosticRoutes.js [hunk 2]
```diff
@@ -20,15 +20,11 @@
 
 const routeResultTypes = require('./routeResultTypes');
 const routeUtils = require('./routeUtils');
+const { version: restVersion } = require('../../package.json');
+const { version: sdkVersion } = require('../../../catapult-sdk/package.json');
```
Comment (Jaguar0625) [inline rest/src/routes/diagnosticRoutes.js]: clever 👍

File: rest/src/routes/diagnosticRoutes.js [hunk 3]
```diff
@@ -39,5 +35,24 @@ module.exports = {
 				next();
 			});
 		});
+
+		server.get('/diagnostic/serverInfo', (req, res, next) => {
```
Comment (Jaguar0625) [inline rest/src/routes/diagnosticRoutes.js]: would probably prefer dropping info as /server and /storage seem more consistetn

File: rest/test/routes/diagnosticRoutes_spec.js [hunk 4]
```diff
@@ -21,11 +21,72 @@
...
+			const next = () => {};
+			const res = {
+				send: response => {
+					sentResponse = response;
+				},
+				redirect: () => {
+					next();
+				}
+			};
+			// Act:
```
Comment (Jaguar0625) [inline rest/test/routes/diagnosticRoutes_spec.js]: add newline

File: rest/test/routes/diagnosticRoutes_spec.js [hunk 5]
```diff
@@ -21,11 +21,72 @@
...
+			let sentResponse;
+			const next = () => {};
+			const res = {
+				send: response => {
+					sentResponse = response;
+				},
+				redirect: () => {
+					next();
+				}
+			};
```
Comment (Jaguar0625) [inline rest/test/routes/diagnosticRoutes_spec.js]: should really refactor these eventually ^^

File: rest/test/routes/diagnosticRoutes_spec.js [hunk 6]
```diff
@@ -21,11 +21,72 @@
...
+			expect(sentResponse).to.deep.equal({
+				payload: {
+					serverInfo: {
+						restVersion,
+						sdkVersion
+					}
+				},
+				type: 'serverInfo'
+			});
+			done();
```
Comment (Jaguar0625) [inline rest/test/routes/diagnosticRoutes_spec.js]: test is sync so i don't think you need to call done explicitly

File: rest/test/routes/diagnosticRoutes_spec.js [hunk 7]
```diff
@@ -21,11 +21,72 @@
...
+			};
+			// Act:
+			const route = routes[endpointUnderTest];
+			route({}, res, next);
+
+			// Assert:
+			expect(sentResponse).to.deep.equal({
+				payload: {
+					serverInfo: {
+						restVersion,
```
Comment (Jaguar0625) [inline rest/test/routes/diagnosticRoutes_spec.js]: doesn't matter but use same ordering in src + test

File: rest/test/routes/diagnosticRoutes_spec.js [hunk 8]
```diff
@@ -21,11 +21,72 @@
...
+			const routes = {};
+			const server = {
+				get: (path, handler) => {
+					routes[path] = handler;
+				}
+			};
+
+			diagnosticRoutes.register(server, {});
+
+			let sentResponse;
```
Comment (Jaguar0625) [inline rest/test/routes/diagnosticRoutes_spec.js]: sent -> captured ?

File: rest/test/routes/diagnosticRoutes_spec.js (hunk 5 above)
Comment (Vektrat) [reply rest/test/routes/diagnosticRoutes_spec.js]: yes, this is on the list of tech debt i owe you 😅

**Pull request merged**